marquez/client.py

```diff
@@ -0,0 +1,264 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import urllib
+
+
+from marquez.constants import JobRunState
+from marquez.utils import request_utils
+from marquez.utils.request_utils import (InvalidRequestException,
+                                         handle_response)
+
+
+class MarquezClient(object):
+    API_PATH = "api/v1"
+    DEFAULT_TIMEOUT_SEC = 5
+    DEFAULT_TIMEOUT_MILLISECONDS = 1000 * DEFAULT_TIMEOUT_SEC
+    marquez_host = None
+    namespace = None
+
+    dataset_api_client = None
+    datasource_api_client = None
+    jobs_api_client = None
+    namespace_api_client = None
+
+    def __init__(self, host=None, port=None, timeout=None):
+        host = os.environ.get('MARQUEZ_HOST') or host
+        port = os.environ.get('MARQUEZ_PORT') or (
+            str(port) if port else port)
+        t = os.environ.get('MARQUEZ_TIMEOUT') or str(timeout)
+        timeout_milliseconds = (int(t) if t.isnumeric() else
+                                self.DEFAULT_TIMEOUT_MILLISECONDS)
+        self.timeout = float(timeout_milliseconds / 1000)
+
+        if not host or not port:
+            msg = ("Please provide host & port or set the proper env "
+                   "vars: MARQUEZ_HOST, MARQUEZ_PORT")
+            raise ConnectionError(msg)
+
+        logging.info("Connecting to Marquez at {}:{}".format(host, port))
```

Comment:
can we remove all of the logging? or, have a `verbose` flag and only log if `true`

Reply:
Removed except in the case of an error (5xx, 400).

Reply:
can we remove this entirely unless a `verbose` flag is present? the client will be embedded in other systems, and we don't want to unintentionally clutter up any logs